File: 프로그래머스/bjw/멀쩡한 사각형.js
let gcd = (n1, n2) => {
  while (n2 > 0) {
    let r = n1 % n2;
    n1 = n2;
    n2 = r;
  }
  return n1;
};

function solution(w, h) {
  let answer = 0,
    gcd_num = gcd(w, h);
  answer = (w - 1) * (h - 1) + (gcd_num - 1);
  return answer;
}

File: 프로그래머스/jsh/python/멀쩡한 사각형.py
import math
# 최대공약수 이용
# 선이 정수인 좌표를 지날때 각각 width, height를 gcd로 나눈 값의 배수가되는 특징
def solution(w,h):
    return (w*h) - (w+h - math.gcd(w,h))